auth/auth.go

```diff
@@ -172,6 +178,58 @@ func (c *Client) Token(ctx context.Context, cfg *config.MountConfig) (*oauth2.To
 	return &oauth2.Token{AccessToken: gcpSAResp.GetAccessToken()}, nil
 }
 
+func (c *Client) gkeWorkloadIdentity(ctx context.Context, cfg *config.MountConfig) (string, string, error) {
+	// Determine Workload ID parameters from the GCE instance metadata.
+	projectID, err := c.MetadataClient.ProjectID()
+	if err != nil {
+		return "", "", fmt.Errorf("unable to get project id: %w", err)
+	}
+	idPool := fmt.Sprintf("%s.svc.id.goog", projectID)
+
+	clusterLocation, err := c.MetadataClient.InstanceAttributeValue("cluster-location")
+	if err != nil {
+		return "", "", fmt.Errorf("unable to determine cluster location: %w", err)
+	}
+	clusterName, err := c.MetadataClient.InstanceAttributeValue("cluster-name")
+	if err != nil {
+		return "", "", fmt.Errorf("unable to determine cluster name: %w", err)
+	}
+	idProvider := fmt.Sprintf("https://container.googleapis.com/v1/projects/%s/locations/%s/clusters/%s", projectID, clusterLocation, clusterName)
+
+	return idPool, idProvider, nil
+}
+
+func (c *Client) fleetWorkloadIdentity(ctx context.Context, cfg *config.MountConfig) (string, string, error) {
+	const envVar = "GOOGLE_APPLICATION_CREDENTIALS"
```

Comment:
I usually know of `GOOGLE_APPLICATION_CREDENTIALS` in the context of application default credentials where they are consumed by the go client libraries. It seems here that we're just using it to get `idPool` and `idProvider` variables.

Are there other systems that consume this file in this way? I'm wondering if it wouldn't just be better either:

1) make those parameters configurable in individual secret-provider-classes like:

```
apiVersion: secrets-store.csi.x-k8s.io/v1
kind: SecretProviderClass
metadata:
  name: app-secrets
spec:
  provider: gcp
  parameters:
    auth: pod-adc
    identity-pool: $PROJECT_ID.svc.id.goog
    identity-provider: $PROJECT_ID/locations/global/memberships/cluster1
    secrets: |
      ...
```

2) make add a command line flags when deploying the provider like:

```
--fleet-identity-audience=identitynamespace:$PROJECT_ID.svc.id.goog:https://gkehub.googleapis.com/projects/$PROJECT_ID/locations/global/memberships/cluster1
```

It seems weird to me to require a configmap and a credential `/var/run/secrets/tokens/gcp-ksa/token` when the credential itself is never actually used.

Reply:
Thank you for the alternatives, they all would work.
Passing the GOOGLE_APPLICATION_CREDENTIALS via environment variable is the way to make Cloud SDK libraries successfully authenticate from arbitrary customer application code (the library looks for the environment variable).
In our case, we are not using Cloud SDK library to implement authentication (which would be possible, I think, but a much more invasive change). Hence, other options are also feasible and are less verbose and more convenient, I agree.
Passing GOOGLE_APPLICATION_CREDENTIALS as environment variable looked more compatible with the client SDK libraries.

Reply:
Okay I think this makes sense, It also seems like the `GOOGLE_APPLICATION_CREDENTIALS` stuff would all be required if the user for some reason wanted to use `auth: provider-adc` to be the CSI driver pod workload identity.

Reply:
Thank you Tommy, please let me know in which direction should I help with this PR.